### DISCUSSION TITLE
1.3 -> 1.4: session.execute bulk insert produces many INSERT's with ;

### DISCUSSION THREAD

**Discussion by tyhoff** [sqlalchemy/sqlalchemy#7879] [Usage Questions]
I might be going crazy, but I wanted to confirm something that was unexpected in our update from 1.3 to 1.4. We just finished a large-scale update (277 files changed!) and finally merged / deployed it yesterday to great success except we noticed that all our our `INSERT INTO ... VALUES (...)` were split up into separate `INSERT` calls separated by semicolons, which caused our databases to be a little less happy.

Engine execution options are `{"executemany": "values"}`

```
stmt = sqlalchemy.dialects.postgresql.insert(models.DeviceCheckin.__table__).on_conflict_do_nothing()
db.session.execute(stmt, row_dicts)
```

1.3
```
INSERT INTO device_checkins VALUES ( <all rows> );
```

1.4
```
INSERT INTO device_checkins (....); INSERT INTO device_checkins (....); INSERT INTO ...
```

Confirmed it to my 1.4 change by testing the before and after change.

The way to fix this for 1.4 that I've found is to instead place `row_dicts` into the `.values(row_dicts)` on the insert statement directly.

Is this change expected, and if not, is there possibly somewhere else I should look to see why this change took place?

Flask-sqlalchemy on Master
SQLAlchemy 1.4.32
Postgres 13
psycopg2 2.8.4 (updated to 2.9.x and same behavior).

**Comment by zzzeek** [comment]
not expexcted at all, can you please post a bug report with a proper MCVE including sample model, sample statement, sample data?  thanks.

**Reply by zzzeek** [reply]
OK, I can see it doing this, this definitely has to do with the Postgresql fast execution helpers, which have changed in 1.4, but should still be working, so likely another regression.  #7880

**Reply by zzzeek** [reply]
it's the on conflict part.  the "executemanymode values" thing isn't needed also, it's automatic now.

**Reply by zzzeek** [reply]
OK this is actually intentional at the moment, due to #6581.  I can maybe adjust it for "ON CONFLICT DO NOTHING" but there are claimed to be issues when you have an on conflict with more going on inside of it.
